### PR TITLE
Initial Setup of About Me Servlet 

### PR DESCRIPTION
**Pull request by Aimaanhasan**

### CODE REVIEW

**Review by waahm7 — COMMENTED** [review]
Great work! I checked out your branch in Intelij Idea and it is giving me an error that Jsoup is not imported in AboutMeServlet.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
   public Datastore() {
     datastore = DatastoreServiceFactory.getDatastoreService();
+
```
Comment (raufrajar) [inline src/main/java/com/google/codeu/data/Datastore.java]: nit: empty line

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+
+    response.setContentType("text/html");
+
+    String user = request.getParameter("user");
+
+    if(user == null || user.equals("")) {
+      // Request is invalid, return empty response
```
Comment (raufrajar) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java:41]: Probably better to use isEmpty here

Comment (raufrajar) [reply src/main/java/com/google/codeu/servlets/AboutMeServlet.java:41]: Also, might be better to print something for user: response.getOutputStream().println("Invalid user");

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      return;
+    }
+
+    User userData = datastore.getUser(user);
+
+    if(userData == null || userData.getAboutMe() == null) {
+      return;
```
Comment (raufrajar) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java:50]: Better to handle the error case.

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    response.getOutputStream().println(userData.getAboutMe());
```
Comment (raufrajar) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java:53]: String aboutMe = "";
if(userData != null && userData.getAboutMe() != null) {
      aboutMe = userData.getAboutMe();
}
println("About Me: " + aboutMe);

Comment (raufrajar) [reply src/main/java/com/google/codeu/servlets/AboutMeServlet.java:53]: just a thought, it looks fine as is.

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+
+    String userEmail = userService.getCurrentUser().getEmail();
+	String aboutMe = Jsoup.clean(request.getParameter("about-me"), Whitelist.none());
```
Comment (raufrajar) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java]: Great! Always should do sanitization for strings given by user.

**Review by raufrajar — DISMISSED** [review]
Looks great! minor things need to be addressed such as handling of errors.

**Review by waahm7 — APPROVED** [review]
Great job!

**Pull request merged**